src/oncall/api/v0/populate.py

```diff
@@ -35,48 +32,17 @@ def on_post(req, resp, schedule_id):
     # TODO: add images to docstring because it doesn't make sense
     data = load_json_body(req)
     start_time = data['start']
-    start_dt = datetime.fromtimestamp(start_time, utc)
-    start_epoch = epoch_from_datetime(start_dt)
-
-    # Get schedule info
-    schedule = get_schedules({'id': schedule_id})[0]
-    role_id = schedule['role_id']
-    team_id = schedule['team_id']
-    roster_id = schedule['roster_id']
-    first_event_start = min(schedule['events'], key=lambda x: x['start'])['start']
-    period = get_period_len(schedule)
-    handoff = start_epoch + timedelta(seconds=first_event_start)
-    handoff = timezone(schedule['timezone']).localize(handoff)
-
-    # Start scheduling from the next occurrence of the hand-off time.
-    if start_dt > handoff:
-        start_epoch += timedelta(weeks=period)
-        handoff += timedelta(weeks=period)
-    if handoff < utc.localize(datetime.utcnow()):
-        raise HTTPBadRequest('Invalid populate request', 'cannot populate starting in the past')
-
-    check_team_auth(schedule['team'], req)
 
     connection = db.connect()
     cursor = connection.cursor(db.DictCursor)
-
-    future_events, last_epoch = calculate_future_events(schedule, cursor, start_epoch)
-    set_last_epoch(schedule_id, last_epoch, cursor)
-
-    # Delete existing events from the start of the first event
-    future_events = [filter(lambda x: x['start'] >= start_time, evs) for evs in future_events]
-    future_events = filter(lambda x: x != [], future_events)
-    if future_events:
-        first_event_start = min(future_events[0], key=lambda x: x['start'])['start']
-        cursor.execute('DELETE FROM event WHERE schedule_id = %s AND start >= %s', (schedule_id, first_event_start))
-
-    # Create events in the db, associating a user to them
-    for epoch in future_events:
-        user_id = find_least_active_available_user_id(team_id, role_id, roster_id, epoch, cursor)
-        if not user_id:
-            continue
-        create_events(team_id, schedule['id'], user_id, epoch, role_id, cursor)
-
-    connection.commit()
+    cursor.execute('SELECT `scheduler`.`name` FROM `schedule` JOIN `scheduler` ON `schedule`.`scheduler_id` = `scheduler`.`id` WHERE `schedule`.`id` = %s', schedule_id)
+    if cursor.rowcount == 0:
+        raise HTTPNotFound()
+    scheduler_name = cursor.fetchone()['name']
+    s = importlib.import_module('oncall.scheduler.' + scheduler_name)
```

Comment:
should probably externalize importing the scheduler to some helper function somewhere, and maintain a dict of scheduler names -> scheduler cls to avoid needing to importlib multiple times